src/expr/symbol_table.cpp

```diff
@@ -87,12 +88,59 @@ class SymbolTable::Implementation {
 
   /** A set of defined functions. */
   CDHashSet<Expr, ExprHashFunction>* d_functions;
+ // for operator overloading
+ private:
```

Comment:
I would argue to keep this as a matter of style.

I prefer breaking up classes into logical blocks using additional "private:"/"public:" annotations.  In this case this specifies that the next set of functions are related to operator overloading.

I suppose the advantage of grouping all public/private functions is that the interface is all in one place.

However, the advantage of grouping into multiple public/private blocks is that the code is easier to understand and manipulate.  For example, say we later want to remove support for operator overloading from this class.  Then all functions involved are in one contiguous private/public block instead of mixed with the other functions.  This is a bigger advantage, in my opinion.

Reply:
I disagree. Grouping public sections together and early in the class definition optimizes for the "readability" of the code. It focuses the reader on what the contracts are and depending on their task: what they get from the class, what they need to maintain or the scope of what they would have to modify if they change it. My view is that optimizing for readability end up decreasing maintenance costs in the long term, which seems to be your main concern.

But, ultimately "readability" is a matter of group convention. So this is a matter of choosing a convention and sticking to it as a group. We should be arguing these points in the style guide and not on the commit.
https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#declaration-order
Specifically this sentence, which discourages such groupings:
> Alternation between sections should be minimized as much as possible.